flake8-coding fails on python 3.7 with "RuntimeError: generator raised StopIteration"

With [PEP479](https://www.python.org/dev/peps/pep-0479/) enforced on Python 3.7, raising `StopIteration` inside a generator causes a `RuntimeError`.
Simply return instead, and the generator will be terminated.
Caused by
https://github.com/tk0miya/flake8-coding/blob/3af44adf226e52677f72466e92e5490c36e1243b/flake8_coding.py#L55
Full traceback:
```
Traceback (most recent call last):
  File "/sec/Dev/BrainBot/raiden/venv/lib/python3.7/site-packages/flake8_coding.py", line 55, in run
    raise StopIteration()
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/usr/lib/python3.7/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/sec/Dev/BrainBot/raiden/venv/lib/python3.7/site-packages/flake8/checker.py", line 648, in _run_checks
    return checker.run_checks()
  File "/sec/Dev/BrainBot/raiden/venv/lib/python3.7/site-packages/flake8/checker.py", line 579, in run_checks
    self.run_ast_checks()
  File "/sec/Dev/BrainBot/raiden/venv/lib/python3.7/site-packages/flake8/checker.py", line 493, in run_ast_checks
    for (line_number, offset, text, check) in runner:
RuntimeError: generator raised StopIteration
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/sec/Dev/BrainBot/raiden/venv/bin/flake8", line 11, in <module>
    sys.exit(main())
  File "/sec/Dev/BrainBot/raiden/venv/lib/python3.7/site-packages/flake8/main/cli.py", line 16, in main
    app.run(argv)
  File "/sec/Dev/BrainBot/raiden/venv/lib/python3.7/site-packages/flake8/main/application.py", line 396, in run
    self._run(argv)
  File "/sec/Dev/BrainBot/raiden/venv/lib/python3.7/site-packages/flake8/main/application.py", line 384, in _run
    self.run_checks()
  File "/sec/Dev/BrainBot/raiden/venv/lib/python3.7/site-packages/flake8/main/application.py", line 310, in run_checks
    self.file_checker_manager.run()
  File "/sec/Dev/BrainBot/raiden/venv/lib/python3.7/site-packages/flake8/checker.py", line 319, in run
    self.run_parallel()
  File "/sec/Dev/BrainBot/raiden/venv/lib/python3.7/site-packages/flake8/checker.py", line 288, in run_parallel
    for ret in pool_map:
  File "/usr/lib/python3.7/multiprocessing/pool.py", line 354, in <genexpr>
    return (item for chunk in result for item in chunk)
  File "/usr/lib/python3.7/multiprocessing/pool.py", line 748, in next
    raise value
RuntimeError: generator raised StopIteration
make: *** [Makefile:41: lint] Error 1
```

--- a/flake8_coding.py
+++ b/flake8_coding.py
@@ -48,23 +48,23 @@ class CodingChecker(object):
 
     def run(self):
         try:
-                # PEP-263 says: a magic comment must be placed into the source
-                #               files either as first or second line in the file
-                lines = self.read_headers()
-                if len(lines) == 0:
-                    raise StopIteration()
+            # PEP-263 says: a magic comment must be placed into the source
+            #               files either as first or second line in the file
+            lines = self.read_headers()
+            if len(lines) == 0:
+                return
 
-                for lineno, line in enumerate(lines, start=1):
-                    matched = re.search('coding[:=]\s*([-\w.]+)', line, re.IGNORECASE)
-                    if matched:
-                        if self.encodings:
-                            if matched.group(1).lower() not in self.encodings:
-                                yield lineno, 0, "C102 Unknown encoding found in coding magic comment", type(self)
-                        else:
-                            yield lineno, 0, "C103 Coding magic comment present", type(self)
-                        break
-                else:
+            for lineno, line in enumerate(lines, start=1):
+                matched = re.search('coding[:=]\s*([-\w.]+)', line, re.IGNORECASE)
+                if matched:
                     if self.encodings:
-                        yield 0, 0, "C101 Coding magic comment not found", type(self)
+                        if matched.group(1).lower() not in self.encodings:
+                            yield lineno, 0, "C102 Unknown encoding found in coding magic comment", type(self)
+                    else:
+                        yield lineno, 0, "C103 Coding magic comment present", type(self)
+                    break
+            else:
+                if self.encodings:
+                    yield 0, 0, "C101 Coding magic comment not found", type(self)
         except IOError:
             pass

